code/game/objects/buckling.dm

```diff
@@ -4,40 +4,63 @@
 	var/can_buckle = 0
 	var/buckle_lying = -1 //bed-like behaviour, forces mob.lying = buckle_lying if != -1 //except -1 actually means "rotate 90 degrees to the left" as it is used by 1*buckle_lying.
 	var/buckle_requires_restraints = 0 //require people to be handcuffed before being able to buckle. eg: pipes
-	var/mob/living/buckled_mob = null
 	var/buckle_offset = 0
+	var/list/mob/living/buckled_mobs = list()
+	var/max_buckled_mobs = 1
 
 
 //Interaction
 /atom/movable/attack_hand(mob/living/user)
 	. = ..()
-	if(can_buckle && buckled_mob)
-		return user_unbuckle_mob(user)
+	if(can_buckle && buckled_mobs.len)
+		if(buckled_mobs.len > 1)
+			var/unbuckled = input(user, "Who do you wish to unbuckle?","Unbuckle Who?") as null|mob in buckled_mobs
+			if(user_unbuckle_mob(unbuckled,user))
+				return 1
+		else
+			if(user_unbuckle_mob(buckled_mobs[1],user))
+				return 1
 
 /atom/movable/attack_robot(mob/living/user)
 	. = ..()
-	if(can_buckle && buckled_mob && Adjacent(user)) // attack_robot is called on all ranges, so the Adjacent check is needed
-		return user_unbuckle_mob(user)
+	if(can_buckle && buckled_mobs.len && Adjacent(user)) // attack_robot is called on all ranges, so the Adjacent check is needed
+		if(buckled_mobs.len > 1)
+			var/unbuckled = input(user, "Who do you wish to unbuckle?","Unbuckle Who?") as null|mob in buckled_mobs
+			if(user_unbuckle_mob(unbuckled,user))
+				return 1
+		else
+			if(user_unbuckle_mob(buckled_mobs[1],user))
+				return 1
 
 /atom/movable/MouseDrop_T(mob/living/M, mob/living/user)
 	. = ..()
-	if(can_buckle && istype(M))
-		return user_buckle_mob(M, user)
+	if(istype(M) && user_buckle_mob(M, user))
+		return 1
 
 
 //Cleanup
 /atom/movable/Destroy()
 	. = ..()
-	unbuckle_mob()
+	unbuckle_all_mobs(force=TRUE)
 
 /atom/movable/proc/has_buckled_mobs()
-	if(buckled_mob)
+	if(!buckled_mobs)
+		return FALSE
+	if(buckled_mobs.len)
 		return TRUE
-	return FALSE
 
 //procs that handle the actual buckling and unbuckling
-/atom/movable/proc/buckle_mob(mob/living/M, force = 0)
-	if((!can_buckle && !force)|| !istype(M) || (M.loc != loc) || M.buckled || M.buckled_mob || buckled_mob || (buckle_requires_restraints && !M.restrained()) || M == src)
+/atom/movable/proc/buckle_mob(mob/living/M, force = FALSE, check_loc = TRUE)
+	if(!buckled_mobs)
+		buckled_mobs = list()
+
+	if(!istype(M))
+		return 0
+
+	if(check_loc && M.loc != loc)
+		return 0
+
+	if((!can_buckle && !force) || !istype(M) || (M.loc != loc) || M.buckled || (buckled_mobs.len >= max_buckled_mobs) || (buckle_requires_restraints && !M.restrained()) || M == src)
```

Comment:
Get rid of `(M.loc != loc)` in this check, it's bypassing the `check_loc` argument. Also, did you intend to make it so that mobs with mobs buckled to them can still buckle?